packages/twenty-front/src/modules/activities/calendar/components/Calendar.tsx

```diff
@@ -17,19 +37,109 @@ const StyledContainer = styled.div`
   gap: ${({ theme }) => theme.spacing(8)};
   padding: ${({ theme }) => theme.spacing(6)};
   width: 100%;
+  overflow: scroll;
 `;
 
 const StyledYear = styled.span`
   color: ${({ theme }) => theme.font.color.light};
 `;
 
-export const Calendar = () => {
-  const { records: calendarEvents } = useFindManyRecords<CalendarEvent>({
-    objectNameSingular: CoreObjectNameSingular.CalendarEvent,
-    orderBy: { startsAt: 'DescNullsLast', endsAt: 'DescNullsLast' },
-    useRecordsWithoutConnection: true,
+export const Calendar = ({ entity }: { entity: ActivityTargetableObject }) => {
+  const { enqueueSnackBar } = useSnackBar();
+
+  const { calendarEventsPageState } = useCalendarEventStates({
+    calendarEventScopeId: getScopeIdFromComponentId(entity.id),
+  });
+
+  const [calendarEventsPage, setCalendarEventsPage] = useRecoilState(
+    calendarEventsPageState,
+  );
+
+  const [isFetchingMoreEvents, setIsFetchingMoreEvents] = useState(false);
+
+  const [threadQuery, queryName] =
+    entity.targetObjectNameSingular === CoreObjectNameSingular.Person
+      ? [
+          getTimelineCalendarEventsFromPersonId,
+          'getTimelineCalendarEventsFromPersonId',
+        ]
+      : [
+          getTimelineCalendarEventsFromCompanyId,
+          'getTimelineCalendarEventsFromCompanyId',
+        ];
+
+  const threadQueryVariables = {
+    ...(entity.targetObjectNameSingular === CoreObjectNameSingular.Person
+      ? { personId: entity.id }
+      : { companyId: entity.id }),
+    page: 1,
+    pageSize: TIMELINE_CALENDAR_EVENTS_DEFAULT_PAGE_SIZE,
+  } as GetTimelineCalendarEventsFromPersonIdQueryVariables;
+
+  const {
+    data,
+    loading: firstQueryLoading,
+    fetchMore,
+  } = useQuery(threadQuery, {
+    variables: threadQueryVariables,
+    onError: (error) => {
+      enqueueSnackBar(error.message || 'Error loading event threads', {
+        variant: 'error',
+      });
+    },
   });
 
+  const fetchMoreRecords = async () => {
```

Comment:
I feel like that function could be factorized so we use the same in both EmailThread and Calendar. Wdyt?